### **Performance Metrics On MultiClass Classification**

In [1]:
from sklearn import metrics

In [2]:
C = "Cat"
D = "Dog"
F = "Fox"

The precision for the Cat class is the number of correctly predicted Cat out of the predicted Cat

Precision = TP/(TP+FP)

The recall for Cat is the number of correctly predicted Cat photos out of the number of actual Cat

Recall = TP/(TP+FN)

In [3]:
# True Values
y_true = [C,C,C,C,C,C,F,F,F,F,F,F,F,F,F,F,D,D,D,D,D,D,D,D,D]

# Predicted Values
y_pred = [C,C,C,C,D,F,C,C,C,C,C,C,D,D,F,F,C,C,C,D,D,D,D,D,D]

# print the confusion matrix
print(metrics.confusion_matrix(y_true,y_pred))

# print the precision and recall among other metrics
print(metrics.classification_report(y_true,y_pred,digits=3))

[[4 1 1]
 [3 6 0]
 [6 2 2]]
              precision    recall  f1-score   support

         Cat      0.308     0.667     0.421         6
         Dog      0.667     0.667     0.667         9
         Fox      0.667     0.200     0.308        10

    accuracy                          0.480        25
   macro avg      0.547     0.511     0.465        25
weighted avg      0.581     0.480     0.464        25



**Precision**

Cat = 4/(4+3+6) = 0.308

Dog = 6/(1+6+2) = 0.667

Fox = 2/(1+0+2) = 0.667

**Recall**

Cat = 4/(4+1+1) = 0.667

Dog = 6/(3+6) = 0.667

Fox = 2/(6+2+2) = 0.2

**F1 Score**

2(Precision*Recall)/(Precision+Recall)

Cat = 2*(0.308*0.667)/(0.308+0.667) = .410872/.975 = 0.421

**Support** (number of elements in each class)

Cat = 4+1+1 = 6

Dog = 3+6 = 9

Fox = 6+2+2 = 10